addons/account/account.py

```diff
@@ -347,43 +357,48 @@ def _prepare_bank_account(self, name, company, currency_id):
         }
 
     @api.model
-    def _prepare_bank_journal(self, company, line):
-        '''
-        This function prepares the value to use for the creation of a bank journal created through the wizard of
-        generating COA from templates.
-
-        :param company: company for which the wizard is running
-        :param line: dictionary containing the values encoded by the user related to his bank account with keys 
-            - acc_name (char): name of the bank account
-            - account_type (char): kind of liquidity journal to create. Either 'bank' or 'cash'
-            - currency_id (int): id of the currency related to this account if its different than the company currency (False otherwise)
-        :return: mapping of field names and values
-        :rtype: dict
-        '''
-        # we need to loop to find next number for journal code
-        for num in xrange(1, 100):
-            # journal_code has a maximal size of 5, hence we can enforce the boundary num < 100
-            journal_code = line.get('account_type', 'bank') == 'cash' and 'CSH' or 'BNK'
-            journal_code += str(num)
-            journal = self.env['account.journal'].search([('code', '=', journal_code), ('company_id', '=', company.id)], limit=1)
-            if not journal:
-                break
-        else:
-            raise UserError(_('Cannot generate an unused journal code.'))
+    def create_bank_account(self, vals):
```

Comment:
remove this code, not needed anymore because of this: https://github.com/odoo/odoo/pull/7077/files#r34915488